src/MacroTools/Extensions/RectExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿using MacroTools.Wrappers;
+using static War3Api.Common;
+
+namespace MacroTools.Extensions
+{
+    /// <summary>
+    /// Provides a helpful set of extension methods for <see cref="rect"/>s.
+    /// </summary>
+    public static class RectExtensions
+    {
+        /// <summary>
+        /// Renders all units inside the specified area invulnerable/>
+        /// </summary>
+        /// <param name="area"></param>
+        public static void MakeUnitsInvulnerable(this rect area)
```

Comment:
I think we need a different method name for this since it does more than just make units invulnerable. But it's a good method, and I'll backport it to some of the older Quests I did. Not sure on a name yet...